grpc_client.cpp

```diff
@@ -909,12 +1181,11 @@ bool sql_query_iterator_next(SqlQueryIterator* iterator, int* attnums, Datum* dv
     {
         const Column& column = columns[i];
         const Value& value = column.data();
-
-        elog(DEBUG3, "Processing field %d with %s", i, column.name().c_str());
-
         int attnum = attnums[i];
 
-        ValueToDatum(value, pgtypes[attnum], pgtypmods[attnum], &dvalues[attnum], &nulls[attnum]);
+        elog(DEBUG3, "Processing field %d (attnum %d) with \"%s\"", i, attnum, column.name().c_str());
+
+        ValueToDatum(value, pgtypes[i], pgtypmods[i], &dvalues[attnum], &nulls[attnum]);
```

Comment:
Changed indexes (`i` in place of `attnum`) otherwise types weren't matching actual column types.